src/flexRenderer.js

```diff
@@ -179,27 +180,26 @@ class FlexGridCell extends React.PureComponent {
   }
 }
 
-export const defaultColHeaderRenderer = ({ header, sortOrder, width, ...rest }) => (
-  <ColHeader width={width} {...rest} sortable={header.sortable}>
-    {header.display}
-    {sortOrder === 'asc' ? (
-      <SortIndicator>&#x25b2;</SortIndicator>
-    ) : sortOrder === 'desc' ? (
-      <SortIndicator>&#x25bc;</SortIndicator>
-    ) : null}
-  </ColHeader>
-)
+export const defaultColHeaderRenderer = ({ header, sortOrder, width, ...rest }) => {
+  const gridContext = React.useContext(GridToolsContext);
+  return (
+    <ColHeader width={width} {...rest} {...gridContext.columnHeaderProps} sortable={header.sortable}>
+      {header.display}
+      {sortOrder === 'asc' ? (
+        <SortIndicator>&#x25b2;</SortIndicator>
+      ) : sortOrder === 'desc' ? (
+        <SortIndicator>&#x25bc;</SortIndicator>
+      ) : null}
+    </ColHeader>
+  )
+}
 
 // <SortIndicator className="fa fa-caret-up" aria-hidden="true" />
 // <SortIndicator className="fa fa-caret-down" aria-hidden="true" />
 
-class FlexGridColHeader extends React.PureComponent {
-  render() {
-    // console.log('scroll y is ', this.props.scrollY)
-    const { render = defaultColHeaderRenderer, ...rest } = this.props
-    return render(rest)
-  }
-}
+const FlexGridColHeader = ({ render = defaultColHeaderRenderer, ...rest }) => (
```

Comment:
Is there a specific reason that we are making this a non-pure component? It was a PureComponent before that prevents re-render if props didn't change

Reply:
Hooks cannot be added to class components. And since the function defaultColHeaderRenderer uses hook useContext, its calling component(FlexGridColHeader) has to be a function component.

Reply:
In that case, please preserve the PureComponent behavior by using `React.memo`

Reply:
Got that... Will change that.